### PR TITLE
Moved low-level matrix functions into utils file

### LINKED ISSUE #89: Move low-level matrix functions into utils file.
There are several low-level matrix functions in link_deletion and local_attachment_tensor.

Functions such as _symmetrical_from_upper_triangular and upper_triangular_sparse.

These should be consolidated in one utils file.

### PR DESCRIPTION
**Pull request by vanlankveldthijs**
Resolves #89 

Consolidated low-level matrix-manipulation methods in util/matrix_utils.py

### CODE REVIEW

File: dgl_ptm/dgl_ptm/network/link_deletion.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import dgl
 import torch
 
+import dgl_ptm.util.matrix_utils as matrix_utils
```
Comment (SarahAlidoost) [inline dgl_ptm/dgl_ptm/network/link_deletion.py]:
```suggestion
from dgl_ptm.util import matrix_utils
```

File: dgl_ptm/dgl_ptm/network/local_attachment.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 import torch
 import dgl
 
+import dgl_ptm.util.matrix_utils as matrix_utils
```
Comment (SarahAlidoost) [inline dgl_ptm/dgl_ptm/network/local_attachment.py]:
```suggestion
from dgl_ptm.util import matrix_utils
```

File: dgl_ptm/dgl_ptm/util/matrix_utils.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+def apply_mask_to_sparse_matrix(sparse_matrix, mask):
+    """
+    Apply a mask to a sparse matrix and return an appropriately masked sparse matrix.
+
+    Args:
+        sparse_matrix (dgl.sparse.SparseMatrix): the original sparse matrix.
+        mask (tensor): the mask to be applied.
+    
+    Return (dgl.sparse.SparseMatrix) the masked matrix.
```
Comment (SarahAlidoost) [inline dgl_ptm/dgl_ptm/util/matrix_utils.py]: I noticed that different styles of docstrings are used everywhere in dgl_ptm like [google style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstring i.e. `Args:` whereas the [reST style](https://peps.python.org/pep-0287/) i.e. `param:`. Can we make them consistent in `matrix_utils.py`? The rest can be fixed as a part of #37

**Review by SarahAlidoost — APPROVED** [review]
@vanlankveldthijs the network module is much cleaner now and nice that the function names and documentation are also improved in this pull request, thanks :+1: Just some minor comments.

File: dgl_ptm/dgl_ptm/util/matrix_utils.py (hunk 3 above)
Comment (vanlankveldthijs) [reply dgl_ptm/dgl_ptm/util/matrix_utils.py]: Sure thing. Which style should I use?

Comment (SarahAlidoost) [reply dgl_ptm/dgl_ptm/util/matrix_utils.py]: I assume that we are not using Sphinx to generate documentation, so I vote for google style which is more simple.

**Review by SarahAlidoost — APPROVED** [review]
@vanlankveldthijs thanks for addressing the comments :+1:

**Pull request merged**